### ISSUE TITLE
implemented local Outlier Probabilites (LoOP)

### ISSUE THREAD

**Issue by yzhao062** [yzhao062/Pyod#8]
See  https://dl.acm.org/citation.cfm?id=1646195

**Comment by tomron** [comment]
This is implemented here -https://github.com/vc1492a/PyNomaly

**Comment by yzhao062** [comment]
Exactly. I am aware of the great library implemented by Valentino. However, I think it could be better to include LoOP as part of Pyod, which implies to rewrite the model under the current framework shown in pyod/models/base.py. 

You are more than welcome to see the examples of implemented models and submit a PR for LoOP:

- https://github.com/yzhao062/Pyod/blob/master/pyod/models/abod.py
- https://github.com/yzhao062/Pyod/blob/master/pyod/models/pca.py
- https://github.com/yzhao062/Pyod/blob/master/pyod/models/cblof.py

The key is to inherit from base.py and implement fit() and decision_function().
Cheers.

**Comment by vc1492a** [comment]
Just to add to the discussion, I've been working on adding LoOP
functionality to scikit-learn that can be viewed in this pull request:
https://github.com/scikit-learn/scikit-learn/pull/10121.

The scikit-learn implementation is faster than that of PyNomaly, as it's
able to take advantage of scikit-learn's rapid nereast neighbor algorithms
and other nice features. I'm waiting on core scikit-learn contributors to
provide some feedback and comments before continuing with a lot of further
work, but maybe this would be the best way to integrate LoOP into Pyod
going forward if it is accepted into scikit-learn (more maintainable than
PyNonaly as there are more contributors, it's faster, etc.).

Happy to help with this issue when I'm back from vacation in early July.
Thought the above may add some helpful information.

**Comment by kno10** [comment]
I have opened a pull request scikit-learn/scikit-learn#27337 that adds knn, knn weight, and loop outlier detectors to sklearn.
If you are interested in seeing this included in core sklearn, you could express this by a thumbs-up on the pull request to finally have this included (note that this issue has been open on PyOD since 2018!)
